peerprovider/http.go

```diff
@@ -0,0 +1,26 @@
+package peerprovider
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"golang.org/x/net/context"
+)
+
+type httpPeerProvider struct{}
+
+func (httpPeerProvider) Resolve(ctx context.Context, url *url.URL) ([]string, error) {
+	resp, err := http.Get(url.String())
```

Comment:
Can we use `ctxhttp` here so that it takes the context into account for timeouts and cancellation